examples/randomwalks/randomwalks.py

```diff
@@ -1,106 +1,244 @@
+            if gpt2_tokenizer:
+                sample_str = sample_str.replace("|", "")
 
-    shortest_lengths = torch.tensor(shortest_lengths)
+            # Convert the sample into a list of nodes (default to an unused
+            # integer if the node is not found)
+            sample: List[int] = [char_to_node.get(c, 1000) for c in sample_str]
 
-    def metric_fn(samples):
-        # a measure for an invalid or a not found path
-        infty = 100
-        lengths = []
-        ref_lengths = []
+            # Initialise the specific sample length
+            length: Optional[int] = None
 
-        for s in samples:
-            if gpt2_tokenizer:
-                s = s.replace("|", "")
-
-            s = [char_to_node.get(c, 1000) for c in s]
-            length = None
-            for ix in range(len(s)):
-                # a nonexisting path is taken
-                if s[ix] >= n_nodes or ix > 0 and not adj[s[ix - 1], s[ix]]:
-                    length = infty
+            for node in range(len(sample)):
+                # If an invalid path is taken, set the length to the invalid
+                # path score
+                if sample[node] >= n_nodes or node > 0 and not adjacency_matrix[sample[node - 1], sample[node]]:
+                    length = invalid_path_length
                     break
-                elif s[ix] == 0:
-                    length = ix + 1
+
+                # Otherwise increment the length for each move (where we don't
+                # end up at the goal node)
+                elif sample[node] == 0:
+                    length = node + 1
                     break
 
+            # Catch the case where there are no moves
             if length is None:
-                length = infty
+                length = invalid_path_length
 
-            lengths.append(length)
-            # allows for inorder checking of % optimality
-            ref_lengths.append(shortest_lengths[s[0] - 1])
+            # Store the batch item length & optimal length staring from the
+            # start node
+            sample_lengths.append(length)
+            sample_optimal_lengths.append(shortest_lengths[sample[0] - 1])
 
-        lengths = torch.tensor(lengths, dtype=torch.float)
-        bound_lengths = torch.where(lengths.eq(infty), max_length, lengths).abs()
-        ref_lengths = torch.as_tensor(ref_lengths)
+        lengths: torch.Tensor = torch.tensor(sample_lengths, dtype=torch.float)
+        bound_lengths: torch.Tensor = torch.where(
+            lengths.eq(invalid_path_length),
+            max_length,
+            lengths
+        ).abs()
+        optimal_lengths = torch.as_tensor(sample_optimal_lengths)
 
         return {
-            "lengths": lengths,
-            # percentage-optimal \in (0, 1) when compared to the shortest path
-            "optimality": (max_length - bound_lengths) / (max_length - ref_lengths),
+            "lengths": sample_lengths,
```

Comment:
Revert this to `lengths` instead of `sample_lengths` to avoid type errors (`sample_lengths` is `Long`)